crates/evm/evm/src/executors/mod.rs

```diff
@@ -519,19 +519,18 @@ impl Executor {
         }
 
         // Check the global failure slot.
-        // TODO: Wire this up
-        let legacy = true;
-        if !legacy {
-            if let Some(acc) = state_changeset.get(&CHEATCODE_ADDRESS) {
-                if let Some(failed_slot) = acc.storage.get(&GLOBAL_FAIL_SLOT) {
-                    return failed_slot.present_value().is_zero();
+        if let Some(acc) = state_changeset.get(&CHEATCODE_ADDRESS) {
+            if let Some(failed_slot) = acc.storage.get(&GLOBAL_FAIL_SLOT) {
+                if !failed_slot.present_value().is_zero() {
+                    return false;
+                }
+            }
+            if let Ok(failed_slot) = self.backend().storage_ref(CHEATCODE_ADDRESS, GLOBAL_FAIL_SLOT)
+            {
+                if !failed_slot.is_zero() {
+                    return false;
                 }
             }
-            let Ok(failed_slot) = self.backend().storage_ref(CHEATCODE_ADDRESS, GLOBAL_FAIL_SLOT)
-            else {
-                return false;
-            };
-            return failed_slot.is_zero();
```

Comment:
the point here is that we can skip calling the "failed" function because checking the global slot is enough, so the last is zero should be returned entirely.

this can be a different config, i was thinking:
- assertions_revert = bool = true
  - if false assertions emit a log and set an internal failure flag instead of failing immediately by reverting
- legacy_assertions = bool = false
  - if true disables this block and we have to call "failed()" most of the time, this is useful only to support old dapptools projects

these could also be squashed into one string/enum config 

cc @mattsse @mds1

Reply:
> this can be a different config, i was thinking:
> 
> * assertions_revert = bool = true
>   
>   * if false assertions emit a log and set an internal failure flag instead of failing immediately by reverting
> * legacy_assertions = bool = false
>   
>   * if true disables this block and we have to call "failed()" most of the time, this is useful only to support old dapptools projects

ah got it, yeah, that makes sense

Reply:
If we skip calling the `failed()` function, will this break things for assertion libraries that implement their own assertions? ref discussion in https://github.com/foundry-rs/forge-std/pull/503

I also like the `assertions_revert` naming since it's more clear what the behavior is compared to `legacy_assertions`

Reply:
both [vulcan](https://github.com/nomoixyz/vulcan/blob/7e04abf0edf1993d9e4cf56aa49a98311d6f9e80/src/_internal/Vulcan.sol#L37) and [PRBTest](https://github.com/PaulRBerg/prb-test/blob/39d6925bf5a7475f7e43247bd59a3e080d33200c/src/PRBTest.sol#L117) are setting global failure slot, so those should be fine

Reply:
I've renamed the option for reverting cheats to `assertions_revert` and added `legacy_assertions` flag with discussed behavior.

currently `legacy_assertions` is set to `false` by default

Reply:
@DaniPopes I think we want to check slot even if `legacy_assertions = true` because if `assertions_revert` is set to `false` and test contract doesn't have a `failed()` method then tests would never fail even if assertions are

Reply:
Yeah makes sense, how about check slot (`if !zero return false`) -> `if !legacy_assertions { return true; }` -> call `failed()`

Reply:
yep, already updated it https://github.com/foundry-rs/foundry/blob/9474d64f912fa096aa512f673c5a79b97e2646dd/crates/evm/evm/src/executors/mod.rs#L524-L542